internal/services/containers/kubernetes_nodepool.go

```diff
@@ -1180,6 +1193,11 @@ func FlattenDefaultNodePool(input *[]managedclusters.ManagedClusterAgentPoolProf
 		gpuInstanceProfile = string(*agentPool.GpuInstanceProfile)
 	}
 
+	gpuDriver := ""
+	if agentPool.GpuProfile != nil {
+		gpuDriver = string(*agentPool.GpuProfile.Driver)
```

Comment:
Please use `gpuDriver = string(pointer.From(props.GpuProfile.Driver))` to avoid panic if `agentPool.GpuProfile.Driver` is nil.